circle.yml

```diff
@@ -0,0 +1,37 @@
+version: 2
+jobs:
+  test:
+    docker:
+      - image: cypress/base:10
+    steps:
+      - checkout
+      # restore folders with npm dependencies and Cypress binary
+      - restore_cache:
+          keys:
+            - cache-{{ checksum "package.json" }}
+      # install npm dependencies and Cypress binary
+      # if they were cached, this step is super quick
+      - run:
+          name: Install dependencies
+          command: npm ci
+      - run: npm run cy:verify
+      # save npm dependencies and Cypress binary for future runs
+      - save_cache:
+          key: cache-{{ checksum "package.json" }}
+          paths:
+            - ~/.npm
+            - ~/.cache
+      # set up the devenv/'_build' directory
+      - run:
+          command: npm run reset
+      # create the view and start server before starting tests
+      - run:
+          command: npm run restart
+          background: true
+      - run: sleep 25 && npm run cy:run
```

Comment:
I use a utility script that checks that the server is alive to handle this!

`wait-for.sh`
```sh
wget --waitretry=5 --retry-connrefused -vO- $1 > /dev/null
```

then

```sh
script/wait_for.sh http://localhost:8004/primo-explore/search && npm run cy:run
```

Reply:
i tried using [`wait-on`](https://www.npmjs.com/package/wait-on) very [briefly](https://github.com/bulib/primo-explore-bu/pull/11/commits/19ad801d57ab9398d13fb6af5cb688ec42642f13), but have pushed that off to remove the extra complexity until i get green.

thank you, though, this is definitely better than the manual sleep. 👍